cli/ops/worker_host.rs

```diff
@@ -34,7 +42,159 @@ pub fn init(i: &mut Isolate, s: &State) {
     "host_get_message",
     s.core_op(json_op(s.stateful_op(op_host_get_message))),
   );
-  i.register_op("metrics", s.core_op(json_op(s.stateful_op(op_metrics))));
+}
+
+fn create_web_worker(
+  name: String,
+  global_state: GlobalState,
+  permissions: DenoPermissions,
+  specifier: ModuleSpecifier,
+) -> Result<WebWorker, ErrBox> {
+  let state =
+    State::new_for_worker(global_state, Some(permissions), specifier)?;
+
+  let mut worker =
+    WebWorker::new(name.to_string(), startup_data::deno_isolate_init(), state);
+  let script = format!("bootstrapWorkerRuntime(\"{}\")", name);
+  worker.execute(&script)?;
+
+  Ok(worker)
+}
+
+// TODO(bartlomieju): this function should probably live in `cli/web_worker.rs`
+pub fn run_worker_loop(
+  rt: &mut tokio::runtime::Runtime,
```

Comment:
Why does this function take a tokio runtime? Could it not just called tokio_util::run_basic?

Reply:
WebWorkers will load modules using Tokio runtime before running the loop - so they need to create runtime manually beforehand

Reply:
Is the error you get something like "cannot call tokio::spawn outside of a tokio context" ? I wouldn't think that just creating the runtime is enough to solve that... Also if that's the error I wonder where it originates from if not during the event loop.

Reply:
> Is the error you get something like "cannot call tokio::spawn outside of a tokio context" ? I wouldn't think that just creating the runtime is enough to solve that... Also if that's the error I wonder where it originates from if not during the event loop.

I did not encounter such error here

Reply:
So if you don't pass the runtime through, what is the error you get? `rt` doesn't seem to be used in `run_worker_loop` except by `block_on`.

Reply:
I believe I did that because of `worker_is_ready` helper variable - got a big error when I tried to make `run_worker_loop` a closure that captures it. I'm pretty sure I'll refactor this in the next pass